File: Excel/588 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/588 Minimum Sum Pair.xlsx"
input = read_excel(path, range = "A1:D10")
test  = read_excel(path, range = "F1:F5") %>% select(1) %>% pull() 

process_column = function(column) {
  grid = combn(column, 2) %>% t() %>% data.frame() %>% mutate(sum = X1 + X2) %>% arrange(sum) %>% head(1)
  paste0(grid$X1, ", ", grid$X2)
}

result = map(input, process_column) %>% unlist()

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE

File: Excel/500-599/588/588 Challenge.py
import pandas as pd
from itertools import combinations

path = "588 Minimum Sum Pair.xlsx"
input = pd.read_excel(path, usecols="A:D", nrows=10)
test = pd.read_excel(path, usecols="F", nrows=4).values.flatten().tolist()

def process_column(column):
    comb = pd.DataFrame(combinations(column, 2), columns=['X1', 'X2'])
    min_sum_pair = comb.assign(sum=comb['X1'] + comb['X2']).nsmallest(1, 'sum')
    return f"{min_sum_pair.iloc[0]['X1']}, {min_sum_pair.iloc[0]['X2']}"

result = [process_column(input[col]) for col in input.columns]

print(result == test) # True